godot/Demos/WindGrass/GrassMesh.gd

```diff
@@ -2,16 +2,27 @@ tool
 extends MultiMeshInstance
 
 export var extents := Vector2.ONE
-
+export var spawn_outside_circle:bool := false
+export var radius:float := 12
 
 func _ready() -> void:
 	var rng := RandomNumberGenerator.new()
 	rng.randomize()
+	
+	var theta = 0
+	var increase = 1
+	var center = get_parent().global_transform.origin
+	
 	for i in multimesh.instance_count:
 		var transform := Transform().rotated(Vector3.UP, rng.randf_range(-PI/2, PI/2))
-		
-		var x := rng.randf_range(-extents.x, extents.x)
-		var z := rng.randf_range(-extents.y, extents.y)
+		var x
+		var z
+		if (!spawn_outside_circle):
+			x := rng.randf_range(-extents.x, extents.x)
+			z := rng.randf_range(-extents.y, extents.y)
+		else:
+			x := center.x + (radius + rng.randf_range(0,extents.x) ) * cos(theta)
+			z := center.z + (radius + rng.randf_range(0,extents.y) ) * sin(theta)
```

Comment:
I think the goal is to add `increase` to `theta` at the end of the loop, but you're not doing it here, so theta is always 0 (resulting in a wall of grass instead of a circle.)

Reply:
this has been tested, there's no wall of grass, it works perfectly!

Reply:
![image](https://user-images.githubusercontent.com/2823241/89716676-066a7180-d97d-11ea-89c4-c7c034a41b9f.png)
The top image is the default code. The bottom image is after adding a `theta += increase` line inside of the loop.

There is nothing that changes theta's value inside of the loop - so `cos(theta)` and `sin(theta)` always return 1 and 0 respectively.